### PR TITLE
junbon/multiplier-variable_take_profit

### PR DESCRIPTION
**Pull request by junbon-binary**

### CODE REVIEW

File: lib/Finance/Contract/Longcode.pm [hunk 1]
```diff
@@ -184,9 +188,16 @@ sub shortcode_to_parameters {
 
     return $legacy_params if (not exists Finance::Contract::Category::get_all_contract_types()->{$initial_bet_type} or $shortcode =~ /_\d+H\d+/);
 
-    if ($shortcode =~
+    if ($shortcode =~ /^(MULTUP|MULTDOWN)_(R?_?[^_\W]+)_(\d*\.?\d*)_(\d+)_(\d+)_(\d+)$/) {
```
Comment (tom-binary) [inline lib/Finance/Contract/Longcode.pm]: Special-casing `R_*` symbols seems wrong here - if we want to restrict, that should be done in company code, not a public module.

It also prevents usage with the `1HZ*` symbols.

File: share/longcodes.yml [hunk 2]
```diff
@@ -80,4 +80,6 @@ putspread_intraday_fixed_expiry: "Win up to [_7] [_6] if [_1]'s exit tick is bet
 putspread_daily: "Win up to [_7] [_6] if [_1]'s exit tick is between [_4] and [_5] at [_3]."
 runhigh_tick: "Win payout if [_1] rises successively for [plural,_3,%d tick, %d ticks] after the entry spot."
 runlow_tick: "Win payout if [_1] falls successively for [plural,_3,%d tick, %d ticks] after the entry spot."
+multup_daily: "longcode for multup"
```
Comment (tom-binary) [inline share/longcodes.yml]: Is this placeholder text?

File: lib/Finance/Contract/Longcode.pm [hunk 3]
```diff
@@ -77,7 +81,7 @@ sub shortcode_to_longcode {
 
     return $LONGCODES->{legacy_contract} if $params->{bet_type} eq 'Invalid';
 
-    defined $params->{date_expiry} or defined $params->{duration} or die 'Invalid shortcode. No expiry is specified.';
+    defined $params->{date_expiry} or defined $params->{duration} or die 'Invalid shortcode. No expiry is specified. ' . $shortcode;
```
Comment (torsten-binary) [inline lib/Finance/Contract/Longcode.pm]: is that intended or a debugging left-over?

Comment (junbon-binary) [reply lib/Finance/Contract/Longcode.pm]: I wanted to die with shortcode. It is a more complete error message.

File: lib/Finance/Contract/Longcode.pm (hunk 1 above)
Comment (junbon-binary) [reply lib/Finance/Contract/Longcode.pm]: `(R?_?[^_\W]+)`:

1. `MULTUP_1HZ100V_100_10_1571732611_4725332611` will return `$2 = 1HZ100V`
2. `MULTUP_R_100_100_10_1571732611_4725332611` will return `$2 = R_100`

I am not sure if I get what's wrong with this here.

File: share/longcodes.yml (hunk 2 above)
Comment (junbon-binary) [reply share/longcodes.yml]: Yes, the product team hasn't provided a longcode for multiplier.

**Pull request merged**